create_jenkins_job.py

```diff
@@ -551,6 +551,23 @@ def create_job(os_name, job_name, template_file, additional_dict):
         job_config = expand_template('ci_launcher_job.xml.em', job_data)
         configure_job(jenkins, launcher_job_name, job_config, **jenkins_kwargs)
 
+    # configure the launch test job
+    launcher_job_name = 'test_ci_launcher'
+    if not pattern_select_jobs_regexp or pattern_select_jobs_regexp.match(launcher_job_name):
+        os_specific_data = collections.OrderedDict()
+        for os_name in sorted(os_configs.keys() - launcher_exclude):
+            os_specific_data[os_name] = dict(data)
+            os_specific_data[os_name].update(os_configs[os_name])
+            os_specific_data[os_name]['job_name'] = 'test_ci_' + os_name
+        job_data = dict(data)
+        job_data['ci_scripts_default_branch'] = args.ci_scripts_default_branch
+        job_data['label_expression'] = 'built-in || master'
+        job_data['os_specific_data'] = os_specific_data
+        job_data['cmake_build_type'] = 'None'
+        job_data.update(retention_data_by_job_type(launcher_job_name))
+        job_config = expand_template('ci_launcher_job.xml.em', job_data)
+        configure_job(jenkins, launcher_job_name, job_config, **jenkins_kwargs)
```

Comment:
Given that this code is the same as the above (other than the job prefix), can we combine them?  Something like:

```
for launch_prefix in ('', 'test_'):
    launcher_job_name = prefix + 'ci_launcher'
    if not pattern_select_jobs_regexp or pattern_select_jobs_regexp.match(launcher_job_name):
        os_specific_data = collections.OrderedDict()
        for os_name in sorted(os_configs.keys() - launcher_exclude):
            os_specific_data[os_name] = dict(data)
            os_specific_data[os_name].update(os_configs[os_name])
            os_specific_data[os_name]['job_name'] = prefix + 'ci_' + os_name
        job_data = dict(data)
        job_data['ci_scripts_default_branch'] = args.ci_scripts_default_branch
        job_data['label_expression'] = 'built-in || master'
        job_data['os_specific_data'] = os_specific_data
        job_data['cmake_build_type'] = 'None'
        job_data.update(retention_data_by_job_type(launcher_job_name))
        job_config = expand_template('ci_launcher_job.xml.em', job_data)
        configure_job(jenkins, launcher_job_name, job_config, **jenkins_kwargs)
```

Reply:
Applied nearly verbatim in ad6b945.